src/main/scala/subsystem/HasTiles.scala

```diff
@@ -86,7 +108,68 @@ trait HasTileInterruptSources
   }
 }
 
+/** These are sources of "constants" that are driven into the tile.
+  * 
+  * While they are not expected to change dyanmically while the tile is executing code,
+  * they may be either tied to a contant value or programmed during boot or reset.
+  * They need to be instantiated before tiles are attached within the subsystem containing them.
+  */
+trait HasTileInputConstants extends InstantiatesTiles { this: BaseSubsystem =>
+  /** tileHartIdNode is used to collect publishers and subscribers of hartids. */
+  val tileHartIdNode = BundleBridgeEphemeralNode[UInt]()
+
+  /** tileHartIdNexusNode is a BundleBridgeNexus that collects dynamic hart prefixes,
+    *   orReduces them, and combines the reduction with the static ids assigned to each tile,
+    *   producing a unique, dynamic hartid for each tile.
+    */
```

Comment:
```suggestion
  /** tileHartIdNexusNode is a BundleBridgeNexus that collects dynamic hart "prefixes".
    * 
    *  Each "prefix" input is actually the same full width as the hartId, the expected usage
    *  is that each prefix source would set only some non-overlapping portion of the bits to something
    *  other than zero. 
    *  Thus, the tileHartIdNexusNode orReduces all the incoming "prefix" wires along with the static ids
    *  assigned to each tile via hartIdList, producing a unique, dynamic hartid for each tile.
    */
```

Reply:
I'm not sure if these width statements are correct.

Reply:
You are right about the widths.

But currently (and this is just me preserving the extant behavior) it isn't just the inputs that are registered, but rather the inputs, the or'd value fanning out to the tile, the tile inputs, and the value fanning out within the tile (when `p(HartPrefixKey) == true`).

Reply:
ok, do you want me to edit my suggestion to try to capture that?

Reply:
No thank you, I will do it now as I change the name of the `HartPrefixKey`